src/components/DiffView/index.tsx

```diff
@@ -0,0 +1,33 @@
+/// <reference path="index.d.ts"/>
+
+import * as React from 'react';
+import { Diff, ViewType, parseDiff } from 'react-diff-view';
+export interface Props {
+  diff: string;
+  viewType: ViewType;
+}
+
+const diffViewDefaultProps = {
+  viewType: 'unified' as ViewType,
+};
+
+interface DiffViewSFC extends React.SFC<Props> {
```

Comment:
I had to do this monkey business with the `diffViewDefaultProps` const and this extra interface to get TS to recognize the fact that we're providing a default value for `viewType` via `defaultProps`. Otherwise it doesn't recognize it which causes type problems. I spent quite a bit of time trying to figure this one out, so maybe there's a better solution. I believe this is just needed for an SFC and if we were creating a class it might "just work", and if so maybe that's a reason to use a class even though we don't currently need one.

Reply:
What is SFC?

Reply:
Stateless Functional Component

Reply:
It's from a code snippet I found which explained how to make this work.

Reply:
ok. Can we write something a bit simpler?

```diff
diff --git a/src/components/DiffView/index.tsx b/src/components/DiffView/index.tsx
index 258be44..b9682dd 100644
--- a/src/components/DiffView/index.tsx
+++ b/src/components/DiffView/index.tsx
@@ -1,22 +1,15 @@
 /// <reference path="index.d.ts"/>
 
 import * as React from 'react';
-import { Diff, ViewType, parseDiff } from 'react-diff-view';
-export interface Props {
-  diff: string;
-  viewType: ViewType;
-}
-
-const diffViewDefaultProps = {
-  viewType: 'unified' as ViewType,
-};
+import { Diff, DiffProps, ViewType, parseDiff } from 'react-diff-view';
 
-interface DiffViewSFC extends React.SFC<Props> {
-  defaultProps: typeof diffViewDefaultProps;
+interface Props extends DiffProps {
+  diff: string;
 }
 
-const DiffView: DiffViewSFC = (props) => {
+const DiffView = (props: Props) => {
   const { diff, viewType } = props;
+
   const parsedDiff = parseDiff(diff)[0];
 
   return (
@@ -28,6 +21,4 @@ const DiffView: DiffViewSFC = (props) => {
   );
 };
 
-DiffView.defaultProps = diffViewDefaultProps;
-
 export default DiffView;
```

Reply:
The reason that works (for the `defaultProps` part) is that you're using a `class` instead of a functional component. I commented on this code earlier saying that we didn't need this monkey business if we used a class instead of a function, even though we don't need a class in this case. I gather you're saying that you'd prefer we use a class, even though one isn't needed, to be able to write simpler code?

Reply:
> to be able to write simpler code?

Yep, that sounds better to me.